src/file_pipe_log/pipe.rs

```diff
@@ -272,7 +270,7 @@ impl<F: FileSystem> SinglePipe<F> {
         };
         // File header must be persisted. This way we can recover gracefully if power
         // loss before a new entry is written.
-        new_file.writer.sync()?;
+        new_file.writer.sync();
         self.sync_dir(path_id)?;
```

Comment:
This error needs to be handled carefully now. (e.g. remove the newly created file and make sure the old writer is okay to write again) Better just unwrap it as well.

Reply:
`build_file_writer` above is the same.

Reply:
Made `sync_dir` panic if it fails.

But `build_file_writer` should be fine, right? It is the type of panic this PR trying to avoid (this can be confirmed by `test_no_space_write_error`). If it fails, the new file  won't be used for writing and will be recycled the next time `rotate_impl` is called. So, it already meet your expectation?

Reply:
Added a few more verifications in test_file_rotate_error test, should be able to address your concern? PTAL